tests/dragonfly/replication_test.py

```diff
@@ -2626,3 +2628,73 @@ async def test_replica_of_replica(df_factory):
         await c_replica2.execute_command(f"REPLICAOF localhost {replica.port}")
 
     assert await c_replica2.execute_command(f"REPLICAOF localhost {master.port}") == "OK"
+
+
+@pytest.mark.asyncio
+async def test_replication_timeout_on_full_sync_heartbeat_expiry(
+    df_factory: DflyInstanceFactory, df_seeder_factory
+):
+    # Timeout set to 3 seconds because we must first saturate the socket such that subsequent
+    # writes block. Otherwise, we will break the flows before Heartbeat actually deadlocks.
+    master = df_factory.create(
+        proactor_threads=2, replication_timeout=3000, vmodule="replica=2,dflycmd=2"
+    )
+    replica = df_factory.create()
+
+    df_factory.start_all([master, replica])
+
+    c_master = master.client()
+    c_replica = replica.client()
+
+    await c_master.execute_command("debug", "populate", "100000", "foo", "5000")
+
+    class ExpirySeeder:
+        def __init__(self):
+            self.stop_flag = False
+            self.i = 0
+
+        async def run(self, client):
+            while not self.stop_flag:
+                await client.execute_command(f"SET tmp{self.i} bar{self.i} EX 4")
+                self.i = self.i + 1
+
+        async def wait_until(self, count):
+            while not self.i > count:
+                await asyncio.sleep(0.5)
+
+        def stop(self):
+            self.stop_flag = True
+
+    c_master = master.client()
+    c_replica = replica.client()
+
+    seeder = ExpirySeeder()
+    seeder_task = asyncio.create_task(seeder.run(c_master))
+    await seeder.wait_until(50000)
```

Comment:
if the expire seeder will stop setting here wouldnt the bug will reproduce as well?
do we need to continue sending commands to get to the bug?